### PR TITLE
Typescript-Client -  Handle special characters when computing enum var name

### PR DESCRIPTION
**Pull request by phiz71**
Hello,

This PR adds a new option for typescript clients in order to handle '+' and '-' characters when computing the name of an enum value.

By activating this option, '+' and '-' will be replaced by 'plus_' and 'minus_'.

I decided to add a new option to the generator because I considered this behavior can be a breaking change for users that already rely on the current generation mechanism.
And also because I didn't want to change existing unit tests for typescript-angular and typescript-axios.

Fix https://github.com/OpenAPITools/openapi-generator/issues/5110

Mentioning the technical committee for typescript:
@TiFu, @taxpon, @sebastianhaas, @kenisteward, @Vrolijkx, @macjohnny, @topce, @akehir, @petejohansonxo, @amakhrov, @davidgamero, @mkusaka 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 1]
```diff
@@ -386,6 +390,9 @@ public void processOpts() {
         if (additionalProperties.containsKey(CodegenConstants.ENUM_PROPERTY_NAMING)) {
             setEnumPropertyNaming((String) additionalProperties.get(CodegenConstants.ENUM_PROPERTY_NAMING));
         }
+        if (additionalProperties.containsKey(ENUM_PROPERTY_NAMING_REPLACE_SPECIAL_CHAR)) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java:393]: please also add a new `cliOptions.add(new CliOption(` above to make sure this is properly registered and exposed in the docs

Comment (macjohnny) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java:393]: and then please re-generate the docs and commit

Comment (phiz71) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java:394]: Hello @macjohnny, it's done 😃

**Review by macjohnny — COMMENTED** [review]
Are the changes to csharp intentional?

**Comment by phiz71** [comment]
Not at all 🤔 ! Could it be a non-idempotent generation? Because it seems that the changes in those files are just a matter of order.

**Comment by phiz71** [comment]
I've solved conflicts and repush

**Comment by TiFu** [comment]
Thanks for the PR!

Maybe 2 small discussion: 
* Is it worth introducing this as a breaking change and avoiding adding yet another configuration option?
* Are there other cases where we replace special characters with an empty string? 
   Might be worthwhile to just fix this in one go, especially if we go the breaking change route

IMO this is a bug as we generate invalid code from a valid spec (see #5110) and thus may justify changing this behavior.

**Comment by macjohnny** [comment]
@TiFu I would argue that since having values like `A+`, `B`, `C` results in valid code, we should avoid a breaking change, but up to you to merge this one

**Review by TiFu — APPROVED** [review]
I think fine to keep it as is as an option given that it would potentially break other generated TS clients.

**Comment by TiFu** [comment]
@phiz71 Thanks for your PR which as been merged into master! :)

**Pull request merged**